test/equality_test.dart

```diff
@@ -175,6 +175,51 @@ main() {
     expect(equality.hash("foo"), isNot(equals(equality.hash("bar"))));
     expect(equality.hash("fÕÕ"), isNot(equals(equality.hash("fõõ"))));
   });
+
+  group("EqualityBy should use a derived value for ", () {
+    final equality1 = new EqualityBy<List<String>, String>(
+        (e) => e.first);
+    final equality2 = new EqualityBy<List<String>, String>(
+        (e) => e.first, const CaseInsensitiveEquality());
+    final equality3 = new EqualityBy<List<Object>, Object>(
+        (e) => e.first, const IterableEquality());
```

Comment:
I'll make this change, but in the future it's a little unreasonable to leave this nitpick after a month of PR back and forth. It definitely doesn't make it easy to encourage contributions from others.